src/design-system/components/menu-item/MenuItem.tsx

```diff
@@ -0,0 +1,107 @@
+import { FC, useRef } from 'react'
+import { MenuItem as MuiMenuItem } from '@mui/material'
+import { ArrowDropDown, ArrowDropUp, CloseRounded } from '@mui/icons-material'
+
+import { MenuItemProps as CommonMenuItemProps } from './MenuItem.types'
+import { MenuItemColorVariant, MenuItemVariant } from './MenuItem.constants'
+import { cn } from '~/utils/cn'
+
+import '~scss-components/menu-item/MenuItem.scss'
+
+interface MenuItemProps extends CommonMenuItemProps {
+  isDropdown?: boolean
+  density?: 1 | 2
+  isToggled?: boolean
+  onRemove?: () => void
+  variant?: MenuItemVariant
+}
+
+const MenuItem: FC<MenuItemProps> = ({
+  title,
+  additionalInfo,
+  alignVariant = 'left',
+  colorVariant = MenuItemColorVariant.Default,
+  density = 1,
+  graphics,
+  isDropdown = false,
+  isToggled = false,
+  isBottomBorder = false,
+  isDisabled = false,
+  onClick,
+  onRemove,
+  variant = MenuItemVariant.Default
+}) => {
+  const graphicsRef = useRef<HTMLButtonElement | null>(null)
+
+  const handleClick = () => {
+    if (graphicsRef.current) {
+      const interactiveElement =
+        graphicsRef.current.querySelector<HTMLInputElement>(
+          'input[type="radio"], input[type="checkbox"], [role="radio"], [role="checkbox"]'
+        )
+
+      if (interactiveElement) {
+        interactiveElement.click()
+        return
+      }
+    }
+
+    onClick?.()
+  }
+
+  const handleGraphicsClick = (event: React.MouseEvent) => {
+    event.stopPropagation()
+    onClick?.()
+  }
+
+  const handleRemoveItem = (event: React.MouseEvent) => {
+    event.stopPropagation()
+    onRemove?.()
+  }
+
+  return (
+    <MuiMenuItem
+      className={cn(
+        's2s-item',
+        `s2s-item--density-${density}`,
+        `s2s-item--variant-${variant}`,
+        `s2s-item--color-${colorVariant}`,
+        `s2s-item--align-${alignVariant}`,
+        isToggled && 's2s-item--toggled',
+        isBottomBorder && 's2s-item--bottom-border',
+        isDisabled && 's2s-item--disabled'
+      )}
+      disabled={isDisabled}
+      key={title}
```

Comment:
You don't need to pass `key` if element is not used in `map`

Reply:
I overlooked it when I was copying from the Menu to a separate component. Thanks for catching that